pkg/types/ccipocr3/interfaces.go

```diff
@@ -20,11 +20,9 @@ type MessageHasher interface {
 	Hash(context.Context, Message) (Bytes32, error)
 }
 
-type ExtraDataCodec interface {
-	// DecodeExtraArgs reformat bytes into a chain agnostic map[string]any representation for extra args
-	DecodeExtraArgs(extraArgs Bytes, sourceChainSelector ChainSelector) (map[string]any, error)
-	// DecodeTokenAmountDestExecData reformat bytes to chain-agnostic map[string]any for tokenAmount DestExecData field
-	DecodeTokenAmountDestExecData(destExecData Bytes, sourceChainSelector ChainSelector) (map[string]any, error)
+type AddressCodec interface {
+	AddressBytesToString([]byte) (string, error)
+	AddressStringToBytes(string) ([]byte, error)
```

Comment:
We typically decode addresses based on the source chain, so I think this interface needs to include the chain selector.

You can also use the `UnknownAddress` type to make sure we pass in the correct type. Changing the type would probably let you remove many spots that cast the UnknownAddress type to bytes.